yjit/src/asm/x86_64/tests.rs

```diff
@@ -381,6 +381,9 @@ fn test_test() {
     check_bytes("4885c0", |cb| test(cb, RAX, RAX));
     check_bytes("4885f0", |cb| test(cb, RAX, RSI));
     check_bytes("48f74640f7ffffff", |cb| test(cb, mem_opnd(64, RSI, 64), imm_opnd(!0x08)));
+    check_bytes("48f7464008000000", |cb| test(cb, mem_opnd(64, RSI, 64), imm_opnd(0x08)));
+    check_bytes("48f7c108000000", |cb| test(cb, RCX, imm_opnd(0x08)));
+    //check_bytes("48a9f7ffff0f", |cb| test(cb, RAX, imm_opnd(0x0FFFFFF7)));
```

Comment:
Sometimes there's special encodings for RAX. Should try corner cases like `mov RCX, u16::MAX` and `mov RCX, u32::MAX`.

Reply:
Ah, you're right, it works fine with RCX but not with RAX. So that'll be that problem at least.

Our assembler doesn't like u32::MAX as an imm_opnd, but that makes sense. It would need to be a large negative number instead. And whenever I supply a uimm_opnd, it wants to compare against ECX rather than RCX (32 bits, not 64.) I can't tell if this is intended behaviour or not...

Anyway, I'll keep adding cases and leave anything commented out that I can't get to pass.

Reply:
Makes sense that u32::MAX doesn't work 👍